src/components/network-chart/index.js

```diff
@@ -0,0 +1,235 @@
+            <MarkerTooltipContents
+              marker={marker}
+              threadIndex={this.props.threadIndex}
+            />
+          </Tooltip>
+        ) : null}
+      </section>
+    );
+  }
+}
+
+class NetworkChart extends React.PureComponent<Props> {
+  constructor(props: any) {
+    super(props);
+    (this: any)._renderRow = this._renderRow.bind(this);
+    (this: any)._getMarkerLength = this._getMarkerLength.bind(this);
+    (this: any)._getMarkerPosition = this._getMarkerPosition.bind(this);
+    (this: any)._getMarkerStyling = this._getMarkerStyling.bind(this);
+  }
+
+  /**
+   * Determine the maximum zoom of the viewport.
+   */
+  getMaximumZoom(): UnitIntervalOfProfileRange {
+    const { timeRange: { start, end }, interval } = this.props;
+    return interval / (end - start);
+  }
+
+  _getMarkerPosition(markerStart) {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    return (markerStart - timeRange.start) * 75 / timeRangeTotal;
+  }
+
+  _getMarkerLength(markerDuration) {
+    const timeRange = this.props.timeRange;
+    const timeRangeTotal = timeRange.end - timeRange.start;
+
+    let markerLength = markerDuration / timeRangeTotal * 75;
+
+    if (markerLength < 0.15) {
+      markerLength = 0.15;
+    }
+
+    return markerLength;
+  }
+
+  _getMarkerStyling(marker) {
+    const markerLength = this._getMarkerLength(marker.dur);
+    const markerPosition = this._getMarkerPosition(marker.data.startTime);
+
+    const markerStyling = {
+      width: markerLength + '%',
+      left: markerPosition + '%',
+    };
+    return markerStyling;
+  }
+
+  // Create row with correct details
+  _renderRow(nodeId: any, index: number) {
+    const markersPerRange = this._filterMarkersPerRange(
+      this.props.timeRange,
+      this.props.markers
+    );
+
+    return (
+      <NetworkChartRow
+        marker={markersPerRange[index]}
+        index={index}
+        markerStyle={this._getMarkerStyling(markersPerRange[index])}
+        threadIndex={this.props.threadIndex}
+      />
+    );
+  }
+
+  _filterMarkersPerRange(timeRange: Object, markers: []) {
+    return markers.filter(function(marker) {
+      const updateRange =
+        timeRange.start < marker.data.startTime &&
+        timeRange.end > marker.data.endTime;
+      return updateRange;
+    });
+  }
+
+  render() {
+    const { timeRange, markers } = this.props;
+
+    //eslint-disable-next-line
+    const networkMarkersPerRange = this._filterMarkersPerRange( timeRange, markers );
+
+    if (!networkMarkersPerRange.length) {
+      return <NetworkChartEmptyReasons />;
+    }
+
+    return (
+      <div className="networkChart">
+        <VirtualList
+          className="treeViewBody"
+          items={networkMarkersPerRange}
+          renderItem={this._renderRow}
+          itemHeight={ROW_HEIGHT}
+          columnCount={1}
+          focusable={true}
+          specialItems={[]}
```

Comment:
Note that passing the empty array like this is a performance problem.
The issue is that a new instance is created at each `render` execution. As a result the shallow-equality in `VirtualList` fails and this triggers a rerender of `VirtualList`... every time.

There are 2 possible solutions IMO:
1. make `specialItems` optional in VirtualList. (I'd advise to do this in a separate PR so that it's easier to review)
2. always pass the same "version" of `[]`, for example a constant that you keep as an instance variable or a global in this file.

I'd advise 2 if this is a transitory state (eg we want to support selections in this view), and 1 otherwise.

Reply:
We will, so I go with the second option. This is still on the list of things to do as I need to figure out how this works.